src/app/content/utils/highlightLoadingUtils.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  GetHighlightsRequest,
+  GetHighlightsSourceTypeEnum,
+  Highlight,
+  Highlights,
+} from '@openstax/highlighter/dist/api';
+import omit from 'lodash/fp/omit';
+import { AppServices } from '../../types';
+import { assertDefined } from '../../utils';
+import { maxHighlightsApiPageSize } from '../constants';
+import { addSummaryHighlight, getHighlightLocationFilterForPage } from '../highlights/utils';
+import { getNextPageSources, incrementPage } from '../highlights/utils/paginationUtils';
```

Comment:
it doesn't really make any sense to move logic into this file outside the highlighting folder and then reach into the highlighting folder for a bunch of dependencies. either we should pull out all the common stuff or just leave it in highlighting and mentally justify it by saying study guides are a type of highlight